Remake custom auth UI docs (#2577)
wasp-lang/wasp: web/docs/auth/overview.md

We will also have access to the `user` object in our frontend code, so we can show different UI to logged-in and logged-out users. For example, we can show the user's name in the header alongside a **logout button** or a login button if the user is not logged in.

## Available auth UI

> When you have decided which auth method you want to support, you can also choose how you want to present it to your users.

Reviewer: Slight change in wording:
```suggestion
When you have decided which auth methods you want to support, you can also choose how you want to present the auth to your users.
```

We will also have access to the `user` object in our frontend code, so we can show different UI to logged-in and logged-out users. For example, we can show the user's name in the header alongside a **logout button** or a login button if the user is not logged in.

> ## Available auth UI

Reviewer: ```suggestion
## Different ways to use auth
```

When you have decided which auth methods you want to support, you can also choose how you want to present the authorization flows to your users.

#### Generated components

This is the fastest way to ship, with Wasp generating ready-made components for your app.
They allow for some customization to make them consistent with your app.
You don't need to implement any UI or logic, and they just work.

<LinkGrid
  links={[
    { title: 'Email', link: '/docs/auth/email' },
    { title: 'Username and password', link: '/docs/auth/username-and-pass' },
    { title: 'Social Auth', link: '/docs/auth/social-auth/overview' },
  ]}
/>

#### Make your own UI {#custom-auth-ui}

Wasp is flexible enough to let you completely customize your login and signup interface.

> We give you the login and signup functions, and you decide how and when to call them.

Reviewer: ```suggestion
We give you the auth related functions, and you decide how and when to call them.
```

Since for email, we have more than login and signup functions :)

This allows for total customization of the look-and-feel, and the interaction, but it needs a bit more work.

<LinkGrid
  links={[
    { title: 'Email', link: '/docs/auth/email/create-your-own-ui' },
    { title: 'Username and password', link: '/docs/auth/username-and-pass/create-your-own-ui' },
    { title: 'Social Auth', link: '/docs/auth/social-auth/create-your-own-ui' },
  ]}
/>

:::tip

You don't have to choose one _or_ the other! Mix-and-match, and use what you need in each moment.
For example, you can create a custom signup screen, but use Wasp's generated components for login.

:::

> ##### Custom login and signup actions (Not recommended)

Reviewer: ```suggestion
#### Custom login and signup actions
```

I'd keep the size the same because it now looks like a suboption of the second option and not like the third option. 

I'd also remove the `Not recommended` - I think we explain enough with the big red banner on top to scare away inexperienced users not to use this if they don't really need to.

The previously discussed options should cover the vast majority of cases. But, for the few instances where it is not enough,
you can [create your own signup flows](./advanced/custom-auth-actions.md), with completely custom logic.
This is not recommended, and reserved for advanced use cases.

> Please check first if other Wasp features (mainly [auth hooks](../auth-hooks.md)) can handle your requirements.

Reviewer: ```suggestion
Please check first if other Wasp features (mainly [auth hooks](../auth-hooks.md)) can help you with your use case.
```

#### Email

> If you use [Email](./auth-ui) authentication, the default validations are:

Reviewer: Why do we link to auth ui here, when we mention email?

Author: (For this and all the others) I really don't know, I think some autoformatting changed this(?) I'm not sure, but I'm changing them back

For this to happen:

- you need to define the fields that you want saved in the database,

> - you need to customize the `SignupForm` (in the case of [Email](./auth-ui) or [Username & Password](./username-and-pass) auth)

Reviewer: Here as well, we probably don't want to link to auth UI?

Read more about using the signup actions for:

> - email auth [here](./auth-ui#fields-in-the-email-dict) <!-- TODO: these docs are not great at explaining using signup and login actions: https://github.com/wasp-lang/wasp/issues/1438 -->

Reviewer: One more instance, do we really want to link to auth UI? The comment next to this line mentions this issue: https://github.com/wasp-lang/wasp/issues/1438 do we want to remove this comment now?
